### PR TITLE
[ISSUE #3006]Replace ScheduledExecutorService instead of Timer to avoid affecting other tasks during exception

### PR DESCRIPTION
**Pull request by zhenhe**
When multiple threads are processing timed tasks in parallel, when the Timer runs multiple TimeTasks, as long as one of them does not catch the thrown exception, the other tasks will automatically stop running. Using ScheduledExecutorService does not have this problem.

**Make sure set the target branch to `develop`**

## What is the purpose of the change

Replace ScheduledExecutorService instead of Timer to avoid affecting other tasks during exception

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: example/src/main/java/org/apache/rocketmq/example/benchmark/TransactionProducer.java [hunk 1]
```diff
@@ -17,37 +17,25 @@
...
 import org.apache.commons.cli.Options;
 import org.apache.commons.cli.PosixParser;
+import org.apache.commons.lang3.concurrent.BasicThreadFactory;
 import org.apache.rocketmq.client.exception.MQClientException;
-import org.apache.rocketmq.client.producer.LocalTransactionState;
-import org.apache.rocketmq.client.producer.SendResult;
-import org.apache.rocketmq.client.producer.SendStatus;
-import org.apache.rocketmq.client.producer.TransactionListener;
-import org.apache.rocketmq.client.producer.TransactionMQProducer;
+import org.apache.rocketmq.client.producer.*;
```
Comment (duhenglucky) [inline example/src/main/java/org/apache/rocketmq/example/benchmark/TransactionProducer.java:26]: It is recommended not to merge all imports

**Comment by duhenglucky** [comment]
Submit an issue first and then submit related PR.

File: example/src/main/java/org/apache/rocketmq/example/benchmark/Producer.java [hunk 2]
```diff
@@ -16,32 +16,30 @@
...
 import org.apache.rocketmq.remoting.common.RemotingHelper;
 import org.apache.rocketmq.remoting.exception.RemotingException;
 import org.apache.rocketmq.srvutil.ServerUtil;
 
+import java.io.UnsupportedEncodingException;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.Random;
+import java.util.TimerTask;
+import java.util.concurrent.*;
```
Comment (odbozhou) [inline example/src/main/java/org/apache/rocketmq/example/benchmark/Producer.java:40]: It is recommended not to merge all imports

File: example/src/main/java/org/apache/rocketmq/example/benchmark/TransactionProducer.java (hunk 1 above)
Comment (zhenhe) [reply example/src/main/java/org/apache/rocketmq/example/benchmark/TransactionProducer.java]: pls check

File: example/src/main/java/org/apache/rocketmq/example/benchmark/TransactionProducer.java [hunk 3]
```diff
@@ -17,37 +17,25 @@
...
 import org.apache.rocketmq.common.message.Message;
 import org.apache.rocketmq.common.message.MessageConst;
 import org.apache.rocketmq.common.message.MessageExt;
 import org.apache.rocketmq.srvutil.ServerUtil;
 
+import java.io.UnsupportedEncodingException;
+import java.nio.ByteBuffer;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.concurrent.*;
```
[the remark this answers is not in the document]
Comment (zhenhe) [reply example/src/main/java/org/apache/rocketmq/example/benchmark/TransactionProducer.java]: done

File: example/src/main/java/org/apache/rocketmq/example/benchmark/Producer.java (hunk 2 above)
Comment (zhenhe) [reply example/src/main/java/org/apache/rocketmq/example/benchmark/Producer.java]: done

**Comment by odbozhou** [comment]
Please create an issue first.
Associate the issue number in the pr name, such as this pr: https://github.com/apache/rocketmq/pull/2970

**Comment by zhenhe** [comment]
> Please create an issue first.
> Associate the issue number in the pr name, such as this pr: #2970

Already created，#3001

**Review by caigy — APPROVED** [review]

**Comment by ltamber** [comment]
@odbozhou can you review this pr again ?

**Review by odbozhou — APPROVED** [review]

**Pull request merged**